packages/viewer/src/modules/menu/components/advancedTools/ImportFile/parser/KMLParser.class.ts

```diff
@@ -62,19 +63,22 @@ export class KMLParser extends FileParser<KMLLayer> {
         if (!isKmlFeaturesValid(kmlAsText)) {
             warningMessages.push(
                 new WarningMessage('kml_malformed', {
-                    filename: kmlFileUrl,
+                    filename: kmlName,
                 })
             )
         }
 
         return layerUtils.makeKMLLayer({
+            name: kmlName,
             opacity: 1.0,
             isVisible: true,
             extent: extentInCurrentProjection,
             kmlFileUrl,
             kmlData: kmlAsText,
             extentProjection: currentProjection,
             warningMessages,
+            hasWarning: warningMessages.length > 0,
+            isLoading: false, // Local files already have their data loaded
```

Comment:
Can you move that to the util/helper function itself, and when the KML we want to create has already its data defined, we set this `isLoading` to false?